apps/dex/src/domains/tokenRegistry/hooks/useTokenRegistry.ts

```diff
@@ -25,53 +25,45 @@ export default function useTokenRegistryQuery(
 
   const { indexedBySymbol, ...assetsQuery } = useAssetsQuery(networkKind);
 
-  const entries = useMemo<EnhancedRegitryAsset[]>(() => {
+  const entries = useMemo<EnhancedRegistryAsset[]>(() => {
     if (!data?.registry?.entries || !indexedBySymbol) {
-      return [] as Array<EnhancedRegitryAsset>;
+      return [] as EnhancedRegistryAsset[];
     }
 
-    const { enhancedAssets } = data.registry.entries.reduce(
-      (acc, entry) => {
-        const asset = (indexedBySymbol[entry.denom.toLowerCase()] ||
-          indexedBySymbol[entry.baseDenom.toLowerCase()] ||
-          indexedBySymbol[
-            entry.denom.slice(1).toLowerCase()
-          ]) as EnhancedRegitryAsset;
-
-        if (asset && acc.symbols.indexOf(asset.symbol) === -1) {
-          acc.symbols.push(asset.symbol);
-
-          const chainId =
-            env !== undefined &&
-            entry.denom === env.nativeAsset.symbol.toLowerCase()
-              ? env?.sifChainId
-              : entry.ibcCounterpartyChainId;
-
-          asset.chainId = chainId;
-          asset.denom = entry.denom;
-          acc.enhancedAssets.push(asset);
-        }
-
-        return acc;
-      },
-      {
-        symbols: [],
-        enhancedAssets: [],
-      } as {
-        symbols: string[];
-        enhancedAssets: EnhancedRegitryAsset[];
-      },
+    return (
+      data.registry.entries
+        // base token have no unitDenom
+        .filter((x) => x.unitDenom === "")
+        .reduce((acc, entry) => {
+          const asset = (indexedBySymbol[entry.denom.toLowerCase()] ||
+            indexedBySymbol[entry.baseDenom.toLowerCase()] ||
+            indexedBySymbol[entry.denom.slice(1).toLowerCase()]) as
+            | EnhancedRegistryAsset
+            | undefined;
+
+          return asset === undefined
+            ? acc
+            : [
+                ...acc,
+                {
```

Comment:
😅 that fine line between cleverness and clarity!

Reply:
Yeah, I was initially gonna do this, but honestly combining it into one return doesn't actually affect the clarity in this case imho :v
```ts
if (asset === undefined) return acc;

return [
  ...acc,
  {
    ...asset,
    denom: entry.denom,
    chainId: entry.denom === env?.nativeAsset.symbol.toLowerCase()
      ? enc.sifChainId
      :  entry.ibcCounterpartyChainId
  }
]
```

Reply:
You did mention in another PR that it is all about perspective/coding style.

From perspective, this snippet is an example of over-engineered code. 

The second part where we have `...acc`, adds little to the code itself. We don't need purity in this case because the reduce accumulator is mutable by nature.

There was nothing wrong with the previous snippet:

```js
if (asset) {
  asset.denom = entry.denom;
  // etc
}

return acc;
```

Logic-wise, both are correct. ✅ 

Clarity and readiness-wise? It is a matter of perspective 😆

Reply:
In Javascript, especially Typescript realm though we want to adhere to functional/declarative rather than imperative programming. And one of the core principles of that paradigm is not having mutations.

Actually, in most cases, you should always prefer declarative over imperative.

Reply:
I'll just leave this here:
https://github.com/jfmengels/eslint-plugin-fp